### PR TITLE
Refactors cli.py and adds two new arguments to the list command

### PR DESCRIPTION
**Pull request by Rohan-cod**
What do these changes do?
-------------------------

- Refactored complete code according to the [**google python style guide**](https://google.github.io/styleguide/pyguide.html).

- Added two new arguments, `pattern` and `regex` to the **`list`** command.

### CODE REVIEW

File: python/vineyard/cli.py [hunk 1]
```diff
@@ -16,30 +16,50 @@
...
+    ls_opt.add_argument('--regex',
+                        action='store_true',
+                        help='The pattern string will be considered as a regex expression')
+    ls_opt.add_argument('--limit', default=5, type=int, help='The limit to list')
+
+    get_opt = cmd_parser.add_parser('get', add_help=False, help='Get a vineyard object')
+    get_opt.add_argument('--object_id', required=True, help='Id of the object to be fetched')
+
+    del_opt = cmd_parser.add_parser('del', add_help=False, help='Delete a vineyard object')
+    del_opt.add_argument('--object_id', required=True, help='Id of the object to be fetched')
```
Comment (siyuan0322) [inline python/vineyard/cli.py:56]: 'to be fetched' -> 'to be deleted'.

Comment (siyuan0322) [reply python/vineyard/cli.py:56]: And I think that changing all `Id` to `ID` would be better.

File: python/vineyard/cli.py [hunk 2]
```diff
@@ -48,14 +68,13 @@ def vineyard_argument_parser():
...
+    """Utility to exit the program with help message in case of any error!"""
     optparser.print_help(sys.stderr)
     sys.exit(-1)
 
 
-__vineyard_client = None
-
-
 def connect_vineyard(args):
+    """Utility to create a vineyard clinet using an IPC or RPC socket!"""
```
Comment (siyuan0322) [inline python/vineyard/cli.py:77]: `clinet` -> `client`

Comment (siyuan0322) [reply python/vineyard/cli.py:77]: And those doc string should not end with a `!`... It's too aggressive. A period will do.

File: python/vineyard/cli.py [hunk 3]
```diff
@@ -67,47 +86,62 @@ def connect_vineyard(args):
     else:
         exit_with_help()
 
-    global __vineyard_client
-    __vineyard_client = client
```
Comment (siyuan0322) [inline python/vineyard/cli.py:71]: Are these two lines have some desired side effects? @sighingnow

**Review by siyuan0322 — COMMENTED** [review]
Good work! Just some minor issues.

File: python/vineyard/cli.py (hunk 3 above)
Comment (sighingnow) [reply python/vineyard/cli.py:71]: Its ok to delete those lines.

**Review by sighingnow — APPROVED** [review]
LGTM. Thank you for the effort on vineyard @Rohan-cod !

**Comment by Rohan-cod** [comment]
> LGTM. Thank you for the effort on vineyard @Rohan-cod !

Pleasure working with you and @siyuan0322. 😇

**Pull request merged**